openmrs/apps/reports/HIV_PATIENTS_LOST_TO_FOLLOWUP/sql/dissagregated_by_patient.sql

```diff
@@ -1,16 +1,24 @@
 SELECT
     patient_id as 'ID',
-    'x' as 'Name',
-    'x' as 'Age',
-    'x' as 'Birthdate',
-    'x' as 'Gender',
-    'x' as 'Phone Nr.',
-    'x' as 'Emergency Contact',
-    'x' as 'ANC Nr.',
-    'x' as 'ART Nr.',
-    'x' as 'NIC',
-    'x' as 'Vital Load Exam Date',
-    'x' as 'Vital Load Exam Result',
-    'x' as 'ARV Start Date',
-    'x' as 'Used ARV Protocols whilst on treatment' 
-FROM patient;
\ No newline at end of file
+    getPatientFullName(patient_id) as 'Name',
+    getPatientAge(patient_id) as 'Age',
+    getPatientBirthdate(patient_id) as 'Birthdate',
+    getPatientGender(patient_id) as 'Gender',
+    getPatientPhoneNumber(patient_id) as 'Phone Nr.',
+    getPatientEmergencyContact(patient_id) as 'Emergency Contact',
+    getPatientANCNumber(patient_id) as 'ANC Nr.',
+    getPatientARTNumber(patient_id) as 'ART Nr.',
+    getPatientCNINumber(patient_id) as 'NIC',
+    getViralLoadTestDate(patient_id) as 'Vital Load Exam Date',
+    getViralLoadTestResult(patient_id) as 'Vital Load Exam Result',
+    getPatientARVStartDate(patient_id) as 'ARV Start Date',
+    getListOfActiveARVDrugs(patient_id, '#startDate#', '#endDate#') as 'Used ARV Protocols whilst on treatment' 
+FROM patient
+WHERE
+    patientHasEnrolledIntoHivProgram(patient_id) = "Yes" AND
+    patientHasTherapeuticLine(patient_id, 0) AND
+    patientHasStartedARVTreatmentBefore(patient_id, '#endDate#') AND
+    patientEndOfARVTreatmentIsBefore(patient_id, '#startDate#') AND
```

Comment:
@tmvumbi2 Why check that the ARV treatment ended before the start of the reporting period? The spec says: **Drug must not end before the start of the reporting quarter**

The existing function **patientWasOnARVTreatmentAtEndReportingPeriod** could be used here?

Reply:
Maybe we could rename patientWasOnARVTreatmentAtEndReportingPeriod to patientWasOnARVTreatmentByDate or something like this?